src/Microsoft.AspNetCore.Analyzer.Testing/DiagnosticVerifier.cs

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.DependencyModel;
+using Microsoft.Extensions.DependencyModel.Resolution;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Analyzer.Testing
+{
+    /// <summary>
+    /// Superclass of all Unit Tests for DiagnosticAnalyzers
+    /// </summary>
+    public class DiagnosticVerifier
+    {
+        internal static string DefaultFilePathPrefix = "Test";
```

Comment:
Make this public? The file name's part of the diagnostic, it'll be good to not hardcode this string in test code.